broadcast/internal/arc/arc_query_tx_test.go

```diff
@@ -0,0 +1,96 @@
+package arc
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/bitcoin-sv/go-broadcast-client/broadcast"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestQueryTransaction(t *testing.T) {
+	testCases := []struct {
+		name           string
+		httpResponse   *http.Response
+		httpError      error
+		expectedResult *broadcast.QueryTxResponse
+		expectedError  error
+	}{
+		{
+			name: "successful request",
+			httpResponse: &http.Response{
+				StatusCode: http.StatusOK,
+				Body: io.NopCloser(bytes.NewBufferString(`
+					{
+						"blockHash": "abc123",
+						"txStatus": "CONFIRMED"
+					}
+					`)),
+			},
+			expectedResult: &broadcast.QueryTxResponse{
+				BlockHash: "abc123",
+				TxStatus:  broadcast.Confirmed,
+			},
+		},
+		{
+			name:          "error in HTTP request",
+			httpError:     errors.New("some error"),
+			expectedError: errors.New("some error"),
+		},
+		// TODO: uncomment this test when we have a way to handle this error (API error)
+		// {
+		// 	name: "missing blockHash in response",
+		// 	httpResponse: &http.Response{
+		// 		StatusCode: http.StatusOK,
+		// 		Body: io.NopCloser(bytes.NewBufferString(`
+		// 			{
+		// 				"txStatus": "CONFIRMED"
+		// 			}
+		// 			`)),
+		// 	},
+		// 	expectedError: ErrMissingHash,
+		// },
```

Comment:
Do we have a task for handling such errors?

Reply:
Replaced with valid test